File: app/src/main/java/com/wolcano/musicplayer/music/di/component/ApplicationComponent.java
package com.wolcano.musicplayer.music.di.component;

import android.app.Application;

import com.wolcano.musicplayer.music.App;
import com.wolcano.musicplayer.music.di.module.ApplicationModule;
import com.wolcano.musicplayer.music.di.scope.PerApplication;

import java.lang.annotation.RetentionPolicy;

import dagger.Component;

@PerApplication
@Component(modules ={ApplicationModule.class})
public interface ApplicationComponent {

    Application application();

    App wolcanoApp();

}

File: app/src/main/java/com/wolcano/musicplayer/music/di/component/ApplicationComponent.kt
package com.wolcano.musicplayer.music.di.component

import android.app.Application
import com.wolcano.musicplayer.music.App
import com.wolcano.musicplayer.music.di.module.ApplicationModule
import com.wolcano.musicplayer.music.di.scope.PerApplication
import dagger.Component

@PerApplication
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {
    fun application(): Application?
    fun wolcanoApp(): App?
}